cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedAbstainDRep.hs

```diff
@@ -0,0 +1,357 @@
+  void $ H.execCli' execConfig
+    [ "conway", "stake-address", "vote-delegation-certificate"
+    , "--always-abstain"
+    , "--stake-verification-key-file", vKeyFile
+    , "--out-file", voteDelegationCertificatePath
+    ]
+
+  -- Compose transaction to publish delegation certificate
+  repRegTxBody1 <- createCertificatePublicationTxBody execConfig epochStateView sbe baseDir "del-cert-txbody"
+                                                      (File voteDelegationCertificatePath) payingWallet
+
+  -- Sign transaction
+  repRegSignedRegTx1 <- signTx execConfig cEra baseDir "signed-reg-tx"
+                               repRegTxBody1 [ SomeKeyPair (paymentKeyInfoPair payingWallet)
+                                             , SomeKeyPair skeyPair]
+
+  -- Submit transaction
+  submitTx execConfig cEra repRegSignedRegTx1
+
+  -- Wait two epochs
+  (EpochNo epochAfterProp) <- getCurrentEpochNo epochStateView
+  void $ waitUntilEpoch (File configurationFile) (File socketPath) (EpochNo (epochAfterProp + 2))
+
+desiredPoolNumberProposalTest
+  :: (MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m, Foldable t)
+  => H.ExecConfig
+  -> EpochStateView
+  -> FilePath
+  -> FilePath
+  -> ConwayEraOnwards ConwayEra
+  -> FilePath
+  -> FilePath
+  -> PaymentKeyInfo
+  -> Maybe (String, Word32)
+  -> t (Int, String)
+  -> Integer
+  -> Integer
+  -> Integer
+  -> m (String, Word32)
+desiredPoolNumberProposalTest execConfig epochStateView configurationFile socketPath ceo work prefix
+                              wallet previousProposalInfo votes change expected epochsToWait = do
+  let sbe = conwayEraOnwardsToShelleyBasedEra ceo
+
+  baseDir <- H.createDirectoryIfMissing $ work </> prefix
+
+  let propVotes :: [DefaultDRepVote]
+      propVotes = zip (concatMap (uncurry replicate) votes) [1..]
+  annotateShow propVotes
+
+  thisProposal@(governanceActionTxId, governanceActionIndex) <-
+    makeDesiredPoolNumberChangeProposal execConfig epochStateView (File configurationFile) (File socketPath)
+                                        ceo baseDir "proposal" previousProposalInfo (fromIntegral change) wallet
+
+  voteChangeProposal execConfig epochStateView sbe baseDir "vote"
+                     governanceActionTxId governanceActionIndex propVotes wallet
+
+  (EpochNo epochAfterProp) <- getCurrentEpochNo epochStateView
+  H.note_ $ "Epoch after \"" <> prefix <> "\" prop: " <> show epochAfterProp
+
+  void $ waitUntilEpoch (File configurationFile) (File socketPath) (EpochNo (epochAfterProp + fromIntegral epochsToWait))
+  desiredPoolNumberAfterProp <- getDesiredPoolNumberValue execConfig
+
+  desiredPoolNumberAfterProp === expected
+
+  return thisProposal
+
+makeDesiredPoolNumberChangeProposal
+  :: (H.MonadAssertion m, MonadTest m, MonadCatch m, MonadIO m)
+  => H.ExecConfig
+  -> EpochStateView
+  -> NodeConfigFile 'In
+  -> SocketPath
+  -> ConwayEraOnwards ConwayEra
+  -> FilePath
```

Comment:
Can you add haddocks to the arguments?

Reply:
Fixed here: https://github.com/IntersectMBO/cardano-node/pull/5815/commits/e82f9f3405d9a6e7df8c34fd4eef4ee94d47a11f